File: p543p557editInsertAndRearrangeRows/p543editInsertAndRearrangeRows/MyCell.m

#import "MyCell.h"


@implementation MyCell
@synthesize textField;


- (void) didTransitionToState:(UITableViewCellStateMask)state {
    [super didTransitionToState:state];
    if (state == UITableViewCellStateEditingMask) {
        self.textField.enabled = YES;
    }
    if (state == UITableViewCellStateDefaultMask) {
        self.textField.enabled = NO;
    }
}

@end

File: iOS8bookExamples/bk2ch08p447editInsertAndRearrangeRows/ch21p741editInsertAndRearrangeRows/MyCell.swift

import UIKit

class MyCell : UITableViewCell {
    @IBOutlet weak var textField : UITextField!
    override func didTransitionToState(state: UITableViewCellStateMask) {
        // well, I think I had this wrong all these years
        // but sheesh, Swift numerics are getting harder and harder
        let editing = UITableViewCellStateMask.ShowingEditControlMask.rawValue
        if state.rawValue & editing != 0 {
            self.textField.enabled = true
        }
        else {
            self.textField.enabled = false
        }
    }
}
